frontend/www/js/omegaup/group/edit.ts

```diff
@@ -0,0 +1,281 @@
+            this.refreshMemberList();
+            source.showEditForm = false;
+            source.showChangePasswordForm = false;
+            source.$el.scrollIntoView();
+          },
+          'bulk-identities': (
+            source: group_Identities,
+            identities: types.Identity[],
+          ) => {
+            api.Identity.bulkCreate({
+              identities: JSON.stringify(identities),
+              group_alias: payload.groupAlias,
+            })
+              .then(() => {
+                this.refreshMemberList();
+                this.tab = AvailableTabs.Members;
+                ui.success(T.groupsIdentitiesSuccessfullyCreated);
+              })
+              .catch(function (data) {
+                ui.error(data.error);
+                source.userErrorRow = data.parameter;
+              });
+          },
+          'download-identities': (identities: types.Identity[]) => {
+            const dialect = {
+              dialect: {
+                csvddfVersion: 1.2,
+                delimiter: ',',
+                doubleQuote: true,
+                lineTerminator: '\r\n',
+                quoteChar: '"',
+                skipInitialSpace: true,
+                header: true,
+                commentChar: '#',
+              },
+            };
+            const csv = CSV.serialize(
+              {
+                fields: [
+                  { id: 'username' },
+                  { id: 'name' },
+                  { id: 'password' },
+                  { id: 'country_id' },
+                  { id: 'state_id' },
+                  { id: 'gender' },
+                  { id: 'school_name' },
+                ],
+                records: identities,
+              },
+              dialect,
+            );
+            const hiddenElement = document.createElement('a');
+            hiddenElement.href = `data:text/csv;charset=utf-8,${window.encodeURIComponent(
+              csv,
+            )}`;
+            hiddenElement.target = '_blank';
+            hiddenElement.download = 'identities.csv';
+            hiddenElement.click();
+          },
+          'read-csv': (
+            source: group_Identities,
+            fileUpload: HTMLInputElement,
+          ) => {
+            source.identities = [];
+            CSV.fetch({
+              file: fileUpload.files ? fileUpload.files[0] : null,
+            }).done((dataset: CSV.Dataset) => {
+              if (dataset.fields.length != 6) {
+                ui.error(T.groupsInvalidCsv);
+                return;
+              }
+              for (const cells of dataset.records) {
+                source.identities.push({
+                  username: `${payload.groupAlias}:${cells[0]}`,
+                  name: cells[1],
+                  password: this.generatePassword(),
+                  country_id: cells[2],
+                  state_id: cells[3],
+                  gender: cells[4],
+                  school_name: cells[5],
+                });
```

Comment:
nit:

```suggestion
              for (const [ username, name, country_id, state_id, gender, school_name ] of dataset.records) {
                source.identities.push({
                  username: `${payload.groupAlias}:username}`,
                  name,
                  password: this.generatePassword(),
                  country_id,
                  state_id,
                  gender,
                  school_name,
                });
```

con estos trucos ([Array destructuring](https://www.typescriptlang.org/docs/handbook/variable-declarations.html#array-destructuring) y [Object literal shorthand](https://rules.sonarsource.com/typescript/RSPEC-3498)) queda un poco más entendible y conciso.

Reply:
Listo. Se ve mucho mejor así.